src/cls/queue/cls_queue_src.cc

```diff
@@ -310,11 +312,7 @@ int queue_list_entries(cls_method_context_t hctx, const cls_queue_list_op& op, c
   
     bufferlist bl_chunk;
     //Read chunk size at a time, if it is less than contiguous data size, else read contiguous data size
-    if (contiguous_data_size > chunk_size) {
-      size_to_read = chunk_size;
-    } else {
-      size_to_read = contiguous_data_size;
-    }
+    size_to_read = std::min(contiguous_data_size, large_chunk_size);
```

Comment:
@mattbenjamin , @mkogan1 @ifed01 : are we using 4k chunk sizes to read data or 4M? chunk_size though set to 4k is unused, and the crash that we see is in the subsequent read call with size_to_read set to 4194304

Reply:
With this patch we use 4K for reading the header and up to 4M (depending on actual data chunk size) to read the queue data.

Reply:
Thank you for the clarification. The crash is seen for only one of the queues for the sequence - list, remove, list (it crashes at the read call), this sequence works fine for all the queues that are processed prior to the queue where the crash is observed.